boot/freeldr/freeldr/lib/mm/meminit.c

```diff
@@ -432,8 +432,8 @@ PVOID MmFindLocationForPageLookupTable(PFN_NUMBER TotalPageCount)
         // Continue, if the block is not big enough?
         if (MemoryDescriptor->PageCount < PageLookupTablePages) continue;
 
-        // Continue, if it is not at a higher address than previous address
-        if (MemoryDescriptor->BasePage < PageLookupTableStartPage) continue;
+        // Continue, if it is not at a lower address than previous address
+        if ((PageLookupTableStartPage) && (MemoryDescriptor->BasePage > PageLookupTableStartPage)) continue;
```

Comment:
Well... It's possible to enable 32-bit protected mode without paging. Even more, all MM code is written under that assumption. But in case of long mode it's not true: you need prepared page tables to enter it. So Freeldr creates temporary tables, but maps only 1 Gb of RAM on x64 (identically to kernel scape and to user space). And because of aforementioned assumption MM tries to find most high memory area to place PLT. On x86 it's valid; on x64 it's not, because in case there are more than 1 Gb of RAM this will crash. So I think there isn't any harm of placing PLT in lower physical memory (isn't it?), but another solution would be rewritting Freeldr init code so that it correctly determines amount of RAM available and initializes structures accordingly.